lib/puppet/rest/compression.rb

```diff
@@ -0,0 +1,20 @@
+module Puppet::Rest
+  module Compression
+    # Returns the decompressed response body. Returns the body as-is
+    # if not compressed.
+    # @param [String] content_encoding the encoding of the body
+    # @param [String] body the response body, or a chunk of it
+    def self.decompress(content_encoding, body)
```

Comment:
Does this now need to save the instance of the inflater to correctly handle streams? Or provide a similar yield-to-block interface for decoding chunks?

I *think* that's what's going on in [network/http/compression](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/network/http/compression.rb#L61)

Reply:
Ah, yeah, good point. I guess this was working for me because I never had more than one chunk.

Reply:
At that point it might make sense just to use that class outright.